website/docs/module-type-overrides.md

```diff
@@ -41,6 +41,29 @@ When multiple patterns match the same file, the last pattern takes precedence.
 * `esm` overrides matches files to compile and execute as native ECMAScript modules.
 * `package` resets either of the above to default behavior, which obeys `package.json` `"type"` and `tsconfig.json` `"module"` options.
 
+**Note:** This does not mean that you have to write CommonJS in your source `webpack.config.ts` file
+- here you can continue writing imports and exports in ESM syntax.
```

Comment:
How about wordsmithing this a bit to focus on "syntax" and avoid negatives, so we're saying something like "you can still use ECMAScript import / export syntax in your `.ts` file."

Following that by a link to webpack's documentation page makes sense.

Reply:
What do you think about merging this entire page into the "CommonJS vs native ECMAScript modules" page?  Maybe giving that page a better title, like "Module Formats: CommonJS and ECMAScript modules"?

I'm wondering if we can have a single page that comprehensively describes the differences between CommonJS and ECMAScript modules, explains that you can almost always use `import` and `export` syntax, and explains `moduleTypes` overrides.

Reply:
> How about wordsmithing this a bit to focus on "syntax" and avoid negatives, so we're saying something like "you can still use ECMAScript import / export syntax in your `.ts` file."
> 
> Following that by a link to webpack's documentation page makes sense.

> I'm wondering if we can have a single page that comprehensively describes the differences between CommonJS and ECMAScript modules, explains that you can almost always use `import` and `export` syntax, and explains `moduleTypes` overrides.

Nice, I think these two together makes for a really nice change. Then anyone confused can read everything about the module formats, because the information is right next to it.

Reply:
Ok cool.  If I get to this before you, I think I'll take a stab at some of this refactoring -- pushing my changes to this branch -- and ask you for a review to make sure it's all still intuitive.